### PR TITLE
Move package from web

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | -

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
+    "minimum-stability": "dev",
     "require": {
-        "php": "^7.2"
+        "php": "^7.2",
+        "psr/container": "^1.0",
+        "psr/http-server-middleware": "^1.0",
+        "psr/http-message": "^1.0",
+        "psr/http-server-handler": "^1.0",
+        "yiisoft/strings": "^3.0@dev",
+        "yiisoft/yii-web": "^3.0@dev"
```
Comment (samdark) [inline composer.json]: That creates circular dependency: `yii-web` depends on `auth` and `auth` depends on `yii-web`. I expect trouble with circular breaks that hard to fix.

We have the following used from `yii-web` in this package:

```php
use Yiisoft\Yii\Web\User\IdentityInterface;
use Yiisoft\Yii\Web\User\IdentityRepositoryInterface;
```

How about moving these to this package? These are about auth anyway.

File: composer.json [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
         "wiki": "http://www.yiiframework.com/wiki/",
         "irc": "irc://irc.freenode.net/yii",
-        "source": "https://github.com/yiisoft/_____"
+        "source": "https://github.com/yiisoft/auth"
     },
+    "minimum-stability": "dev",
     "require": {
-        "php": "^7.2"
+        "php": "^7.2",
+        "psr/container": "^1.0",
```
Comment (samdark) [inline composer.json:18]: Is it possible to get rid of this dependency? It's not bad as such, just wondering.

**Review by samdark — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (rustamwin) [reply composer.json]: > How about moving these to this package? These are about auth anyway.

:thinking: I'm thinking about splitting `user` package :)

Comment (samdark) [reply composer.json]: That would cause the same problem and you'll have to split:

1. user
2. session
3. cookie

That's a bit too much, IMO.

Comment (rustamwin) [reply composer.json]: OK. `AuthenticationKeyInterface` as well?

Comment (samdark) [reply composer.json]: No. That is "remember me" and isn't directly connected to authentication itself.

File: src/Method/Composite.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
+<?php
+namespace Yiisoft\Auth\Method;
+
+use Psr\Container\ContainerInterface;
```
Comment (samdark) [inline src/Method/Composite.php:4]: That is still a dependency.

**Comment by samdark** [comment]
👍

**Comment by samdark** [comment]
Looks good overall but I think container dependency should be either added back or really removed (not just from composer.json).

**Comment by samdark** [comment]
Merged. Thank you very much!

**Pull request merged**